{
  try {
    result=callable.call();
  }
 catch (  Throwable t) {
    logger.warn("Uncaught exception on thread {}: {}",Thread.currentThread(),t);
    result=t;
    failure=true;
  }
 finally {
    signalAll();
    onCompletion();
  }
}
